datacompy/legacy.py

```diff
@@ -0,0 +1,928 @@
+#
+# Copyright 2024 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+from enum import Enum
+from itertools import chain
+from typing import Any, Dict, List, Optional, Set, TextIO, Tuple, Union
+from warnings import warn
+
+try:
+    import pyspark
+    from pyspark.sql import functions as F
+except ImportError:
+    pass  # Let non-Spark people at least enjoy the loveliness of the pandas datacompy functionality
+
+
+warn(
```

Comment:
The warning message doesn't seem accurate if I understand correctly? This is in the legacy module so it seems like we should either delete this message from this module or change the string to something warning that we're going to delete legacy functionality in a vx.x.x release.

Reply:
Made a tweak.